web_application/Backend/FlaskApp/app.py

```diff
@@ -1426,16 +1422,24 @@ def combine_datasets(dataset_ids, fetch_dataset):
     combined_data = []
 
     for dataset_id in dataset_ids:
-        dataset = fetch_dataset(dataset_id)  # Function to fetch dataset by ID
+        dataset = fetch_dataset(dataset_id)  # Fetch dataset by ID
 
         if 'data' in dataset:
             df = pd.DataFrame.from_dict(dataset['data'], orient='index')
+
+            # Add 'user_id' and rename 'Unnamed: 0' to 'sample_number'
```

Comment:
I'm not sure here: do we store the sample number in the column 'Unnamed' in our db? If so, we should update it to 'sample_number' or simply use it as index.